COM-788 TB_STAT - Number of new and relapsed TB cases with documented HIV status - Newly Tested Positives

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-788

tmvumbi2 committed:
COM-788 Implement testing indicator 8b
Implements testing indicator 8b

lauravignoli commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -409,6 +409,37 @@ BEGIN
+    WHERE
+        patientGenderIs(pat.patient_id, p_gender) AND
+        patientHasEnrolledIntoTBProgramDuringReportingPeriod(pat.patient_id, p_startDate, p_endDate) AND
+        getObsDatetimeValueInSection(pat.patient_id, uuidHIVTestDate, uuidHIVTestSection) BETWEEN p_startDate AND p_endDate AND
```
@tmvumbi2 This should be deleted as the date is checked in the following function

lauravignoli commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -409,6 +409,37 @@ BEGIN
+        patientGenderIs(pat.patient_id, p_gender) AND
+        patientHasEnrolledIntoTBProgramDuringReportingPeriod(pat.patient_id, p_startDate, p_endDate) AND
+        getObsDatetimeValueInSection(pat.patient_id, uuidHIVTestDate, uuidHIVTestSection) BETWEEN p_startDate AND p_endDate AND
+        patientHIVFinalTestResultIsWithinReportingPeriod(pat.patient_id, "Positive", p_startDate, p_endDate) AND
```
@tmvumbi2 We should confirm with Richard how to check the Final Result Test Date:

obs date or HIV Test Date in that section

lauravignoli committed:
COM-788 Remove duplicate check from ind8b

This check is already done in patientHIVFinalTestResultIsWithinReportingPeriod.
